packages/core/src/Cardano/types/ProtocolParameters.ts

```diff
@@ -24,27 +27,87 @@ export interface ValidityInterval {
   invalidHereafter?: Slot;
 }
 
-export interface ProtocolParametersBabbage {
-  minFeeCoefficient?: number;
-  minFeeConstant?: number;
-  maxBlockBodySize?: number;
-  maxBlockHeaderSize?: number;
-  maxTxSize?: number;
-  stakeKeyDeposit?: number;
-  poolDeposit?: number;
-  poolRetirementEpochBound?: number;
-  desiredNumberOfPools?: number;
-  poolInfluence?: number;
-  monetaryExpansion?: number;
-  treasuryExpansion?: number;
-  minPoolCost?: number;
-  coinsPerUtxoByte?: number;
-  maxValueSize?: number;
-  collateralPercentage?: number;
-  maxCollateralInputs?: number;
-  protocolVersion?: ProtocolVersion;
-  costModels?: CostModels;
-  prices?: Prices;
-  maxExecutionUnitsPerTransaction?: ExUnits;
-  maxExecutionUnitsPerBlock?: ExUnits;
+export interface TxFeePolicy {
+  coefficient: string;
+  constant: number;
 }
+
+export interface SoftforkRule {
+  initThreshold: string;
+  minThreshold: string;
+  decrementThreshold: string;
+}
+
+export type ProtocolParametersByron = {
+  heavyDlgThreshold: string;
+  maxBlockSize: number;
+  maxHeaderSize: number;
+  maxProposalSize: number;
+  maxTxSize: number;
+  mpcThreshold: string;
+  scriptVersion: number;
+  slotDuration: number;
+  unlockStakeEpoch: number;
+  updateProposalThreshold: string;
+  updateProposalTimeToLive: number;
+  updateVoteThreshold: string;
+  txFeePolicy: TxFeePolicy;
+  softforkRule: SoftforkRule;
+};
+
+type NewProtocolParamsInShelley = {
+  minFeeCoefficient: number;
+  minFeeConstant: number;
+  maxBlockBodySize: number;
+  maxBlockHeaderSize: number;
+  stakeKeyDeposit: number;
+  poolDeposit: number | null;
+  poolRetirementEpochBound: number;
+  desiredNumberOfPools: number;
+  poolInfluence: string;
+  monetaryExpansion: string;
+  treasuryExpansion: string;
+  decentralizationParameter: string;
+  minUtxoValue: number;
+  minPoolCost: number;
+  extraEntropy: 'neutral' | string;
+  protocolVersion: ProtocolVersion;
+};
+
+export type ShelleyProtocolParams = Omit<
```

Comment:
Any reason to not `Pick` instead?

Reply:
Just following the consistency of how we define all next protocol params: 
1. omit the obsolete ones
2. add the new ones

Like [here](https://github.com/input-output-hk/cardano-js-sdk/pull/416/files#diff-2fd64a4659e387924f9375a568cbeb7bb58ff9f4baa9d03d35402c27d81d65adR106), and [here](https://github.com/input-output-hk/cardano-js-sdk/pull/416/files#diff-2fd64a4659e387924f9375a568cbeb7bb58ff9f4baa9d03d35402c27d81d65adR112)

Reply:
I think it'd be better to use the most appropriate utility class to be more DRY, however, I can see the appeal of the semantics. I'm happy with either, so let's see what the others think

Reply:
As I remember from our discussion (a week ago) we agreed on this more declarative approach of defining protocol parameters type for each era which represents the composition of the prior era parameterts, considering deprecations + the new ones added by the current era. 
So we can keep track of all protocol parameter updates in time quickly, correct me if I'm missing something 🤔